refinery/core/api.py

```diff
@@ -831,6 +843,54 @@ def get_study_assays(self, request, **kwargs):
             study_uuid=kwargs['study_uuid']
         )
 
+    def handle_incomplete_dataset(self, dataset):
+        """
+        Handle an incomplete DataSet instance when a detail view is
+        accessed.
+        :param dataset: a DataSet object
+        :return: a DataSet object
+        :raises: A TastyPie NotFound Exception if said DataSet
+        doesn't have the proper information associated yet.
+        """
+        if dataset.get_version_details() is not None:
+            return dataset
+
+        else:
+            logger.error(
+                self.make_incomplete_dataset_error_message(dataset.uuid)
+            )
+            raise NotFound(
+                self.make_incomplete_dataset_error_message(dataset.uuid)
+            )
+
+    def handle_incomplete_datasets(self, datasets):
+        """
+        Handle incomplete DataSet instances when a list view is accessed.
+        :param datasets: a list of DataSet objects
+        :return: a list of DataSet objects sans ones that don't have the
+        proper information associated yet.
+        """
+        complete_datasets = []
+
+        for dataset in datasets:
+            if dataset.get_version_details() is None:
```

Comment:
Actually, I would go further and re-use `handle_incomplete_dataset()` in `get_obj_list()` directly:
```python
    for dataset in datasets:
        try:
            complete_datasets.append(handle_incomplete_dataset(dataset))
        except NotFound:
            pass
```
Then move the functionality of `make_incomplete_dataset_error_message()` into `handle_incomplete_dataset()`.
Then remove both `make_incomplete_dataset_error_message()` and `handle_incomplete_datasets()` completely.